Assets/_Functionalities/Wms/Scripts/WMSTileDataLayer.cs

```diff
@@ -0,0 +1,107 @@
+            }
+        }       
+
+        private void Awake()
+        {
+            //Make sure Datasets at least has one item
+            if (Datasets.Count == 0)
+            {
+                var baseDataset = new DataSet()
+                {
+                    maximumDistance = 3000,
+                    maximumDistanceSquared = 1000 * 1000
+                };
+                Datasets.Add(baseDataset);
+            }
+        }
+
+        protected override IEnumerator DownloadDataAndGenerateTexture(TileChange tileChange, Action<TileChange> callback = null)
+        {
+            var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+
+            if (!tiles.ContainsKey(tileKey))
+            {
+                onLogMessage.Invoke(LogType.Warning, "Tile key does not exist");
+                yield break;
+            }
+
+            Tile tile = tiles[tileKey];
+            var bboxValue = $"{tileChange.X},{tileChange.Y},{(tileChange.X + tileSize)},{(tileChange.Y + tileSize)}";
+            string url = wmsUrl.Replace("{0}", bboxValue);
+
+            UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(url);
+            tile.runningWebRequest = webRequest;
+            yield return webRequest.SendWebRequest();
+            tile.runningWebRequest = null;
+            if (webRequest.result != UnityWebRequest.Result.Success)
+            {
+                Debug.LogWarning($"Could not download {url}");
+                RemoveGameObjectFromTile(tileKey);
+                callback(tileChange);
+            }
+            else
+            {
+                Texture texture = ((DownloadHandlerTexture)webRequest.downloadHandler).texture;
+                Texture2D tex = texture as Texture2D;
+                if (tile.gameObject.TryGetComponent<TextureProjectorBase>(out var projector))
+                {
+                    projector.SetSize(tileSize, tileSize, tileSize);
+                    projector.gameObject.SetActive(isEnabled);                    
+                    projector.SetTexture(tex);
+                    //force the depth to be at least larger than its height to prevent z-fighting
+                    DecalProjector decalProjector = tile.gameObject.GetComponent<DecalProjector>();
+                    TextureDecalProjector textureDecalProjector = tile.gameObject.GetComponent<TextureDecalProjector>();
+                    if (ProjectorHeight >= decalProjector.size.z)
+                        textureDecalProjector.SetSize(decalProjector.size.x, decalProjector.size.y, ProjectorMinDepth);
+
+                    //set the render index, to make sure the render order is maintained
+                    textureDecalProjector.SetPriority(renderIndex);
+
+                }
+                ClearPreviousTexture(tile);
+                callback(tileChange);
+            }
+            yield return null;
```

Comment:
waar is deze yield voor nodig?

Reply:
wow das een oude meegekopieerde artefact. heb ze meteen uit alle andere scripts gehaald.